src/main.spec.js

```diff
@@ -112,3 +113,22 @@ test("Checks HTMLElements", () => {
     }
   }
 });
+
+describe("Tests fetch call", () => {
+  beforeEach(() => {
+    fetch.resetMocks();
+  });
+
+  it("mockResponse", () => {
+    fetch.mockResponseOnce(JSON.stringify({ data: "datatest" }));
+
+    //assert on the response
+    fetch("../demo/demo.json").then(res => {
```

Comment:
Basically what we have to do here is:

1. Call `renderUrl('../demo/demo.json')`, although I suggest changing the URL to something non-real (e.g. `/api/fs/`)
2. Expect `renderInput` to be called with the mocked JSON response